app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,26 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
+    
+    if user 
+      @token = user.token
+    else 
+      redirect_to expired_token_path
+    end
+  end
+  
+  def create
+    user = User.where(token: params[:token]).first 
+    
+    if user 
+      user.password = params[:password]
+      user.generate_token 
+      user.save 
```

Comment:
The problem with approach is that if password would invalid, your app would still tell user its been changed. Also 

```ruby
user.password = params[:password]
user.save
```

can be just `user.update(password: params[:password])`